bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/views/report/list/provider/content/TreeCheckerContentProvider.java

```diff
@@ -74,12 +73,14 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
         if (parentElement instanceof String) {
 
             String parent = (String) parentElement;
-            int parentDots = parent.split("\\.").length;
+            int parentLevel = parent.split("\\.").length;
 
             Set<Object> tops = new HashSet<>();
             for (String s : this.reportListView.getReportList().get().getCheckers()) {
-                if (s.startsWith(parent) && !s.equals(parent)) {
-                    tops.add(parent + "." + s.split("\\.")[parentDots]);
+                String[] splitted = s.split("\\.");
+                if (splitted.length > parentLevel - 1 && splitted[parentLevel - 1].equals(parent)
```

Comment:
wouldn't the following condition be equivalent for our entries in the checkerlist?
`if (splitted.length > parentLevel && splitted[parentLevel - 1].equals(parent)) { ... }`
Maybe not, just asking.